File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
import java.util.*;
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet <Integer> set = new HashSet<>();
        int longest = 1;
        if (nums.length == 0) return 0;

        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }

        for (int val : set ) {
            if (!set.contains(val - 1)) {
                int count = 1, x = val;
                while (set.contains(x + 1)) {
                    count++ ;
                    x++;
                }
                longest = Math.max(longest, count) ;
            }
        }
        return longest;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        longest = 0
        num_set = set(nums)

        for num in num_set:
            if (num - 1) not in num_set:
                curr = num
                count = 1  

                while (curr + 1) in num_set:
                    count+=1
                    curr+=1

                longest = max(longest, count)

        return longest      